### DISCUSSION TITLE
LoRA微调时只有第一个loss有数值，后面的loss全为0.0

### DISCUSSION THREAD

**Discussion by yuanye1010** [zai-org/ChatGLM3#568] [Q&A]
如题，怎么解决呢？

**Comment by hhscc-lz** [comment]
同样的问题，解决了吗兄弟

**Comment by YangLei6** [comment]
同样的问题，解决了吗兄弟

**Comment by BXHQu** [comment]
同样的问题，解决了吗，兄弟

**Comment by Xuanaxx** [comment]
同样的问题，解决了吗兄弟

**Comment by YangLei6** [comment]
我用两个不同的环境，有一个是这种情况，有一个不是，是不是环境的原因？

**Reply by Xuanaxx** [reply]
不清楚，我看网上其它项目类似情况（loss只有第一次有值）给了一个解决方案，把模型精度改为bf16。我试了一下，loss有是有了，但是训练结果也不尽如人意（变成复读机了）...

**Reply by Xuanaxx** [reply]
更新一下，学习率调低也能部分解决问题

**Reply by ZhaoyingAC** [reply]
模型精度设置成bf16, 学习率设置成1e-5还是同样的问题。
```
tokenizer = AutoTokenizer.from_pretrained(model_args.model_name_or_path, trust_remote_code=True, use_bfloat16=True)
model = AutoModel.from_pretrained(model_args.model_name_or_path, trust_remote_code=True, use_bfloat16=True).cuda()
```

**Reply by ZhaoyingAC** [reply]
你的python、torch、transformers的版本都是多少

**Reply by Xuanaxx** [reply]
> 你的python、torch、transformers的版本都是多少

python3.10，torch2.1.2，transformers4.36.2

**Comment by ZhaoyingAC** [comment]
peft==0.6.0 解决了，可以参考

**Reply by huilong-chen** [reply]
同样的问题，解决了吗兄弟

**Reply by sofiane-20241050** [reply]
你说到点子上了，解决这个问题了，至于loss有没有下降和收敛就是另一回事

**Comment by fuzhao123232** [comment]
作者说的peft是0.7.1哦，2024/3/5

**Comment by sofiane-20241050** [comment]
You are right, 我前天注意到了，ok

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年03月05日 16:57 |
| 收件人 | THUDM/ChatGLM3 ***@***.***> |
| 抄送人 | sofiane-20241050 ***@***.***>,
Manual ***@***.***> |
| 主题 | Re: [THUDM/ChatGLM3] LoRA微调时只有第一个loss有数值，后面的loss全为0.0 (Discussion #568) |

作者说的peft是0.7.1哦，2024/3/5
